service/graph_test.go

```diff
@@ -0,0 +1,193 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package service
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gonum.org/v1/gonum/graph/simple"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+)
+
+var _ component.Component = &testNode{}
+
+type testNode struct {
+	id          component.ID
+	startErr    error
+	shutdownErr error
+}
+
+func (n *testNode) ID() int64 {
```

Comment:
Unused?

Reply:
This fulfills the `graph.Node` interface, so that we can use `testNode` in the `simple.DirectedGraph`.

Reply:
Add a comment, for someone unfamiliar with that.